### ISSUE TITLE
Cannot build Sockets extension for PHP 8.0.15

### ISSUE THREAD

**Issue by llaville** [mlocati/docker-php-extension-installer#498]
I've just downloaded a fresh copy of `php:8.0-fpm-buster` and try to rebuild all extensions with `docker-php-extension-installer` v1.4.12, and got following output 

I don't think its an issue from your project, but rather to PHP 8.0.15 itself. @mlocati can you confirm please ?

```
#59 [build-version-8015 47/68] RUN install-php-extensions sockets
#59 sha256:b0cf4571e84b316a19d4ffb64f2111955be2f470dc9a6ed4425a3a15312bcf29
#59 0.469 install-php-extensions v.1.4.12
#59 2.771 ### INSTALLING BUNDLED MODULE sockets ###
#59 2.849 Configuring for:
#59 2.849 PHP Api Version:         20200930
#59 2.849 Zend Module Api No:      20200930
#59 2.849 Zend Extension Api No:   420200930
#59 3.993 checking for grep that handles long lines and -e... /bin/grep
#59 3.999 checking for egrep... /bin/grep -E
#59 4.004 checking for a sed that does not truncate output... /bin/sed
#59 4.071 checking for pkg-config... /usr/bin/pkg-config
#59 4.073 checking pkg-config is at least version 0.9.0... yes
#59 4.077 checking for cc... cc
#59 4.135 checking whether the C compiler works... yes
#59 4.221 checking for C compiler default output file name... a.out
#59 4.224 checking for suffix of executables...
#59 4.311 checking whether we are cross compiling... no
#59 4.388 checking for suffix of object files... o
#59 4.444 checking whether we are using the GNU C compiler... yes
#59 4.492 checking whether cc accepts -g... yes
#59 4.543 checking for cc option to accept ISO C89... none needed
#59 4.612 checking how to run the C preprocessor... cc -E
#59 4.708 checking for icc... no
#59 4.728 checking for suncc... no
#59 4.752 checking for system library directory... lib
#59 4.753 checking if compiler supports -Wl,-rpath,... yes
#59 4.869 checking build system type... x86_64-pc-linux-gnu
#59 4.889 checking host system type... x86_64-pc-linux-gnu
#59 4.891 checking target system type... x86_64-pc-linux-gnu
#59 4.980 checking for PHP prefix... /usr/local
#59 4.981 checking for PHP includes... -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib
#59 4.981 checking for PHP extension directory... /usr/local/lib/php/extensions/no-debug-non-zts-20200930
#59 4.982 checking for PHP installed headers prefix... /usr/local/include/php
#59 4.983 checking if debug is enabled... no
#59 5.018 checking if zts is enabled... no
#59 5.054 checking for gawk... no
#59 5.056 checking for nawk... nawk
#59 5.057 checking if nawk is broken... no
#59 5.061 checking whether to enable sockets support... yes, shared
#59 5.072 checking for hstrerror... yes
#59 5.177 checking for if_nametoindex... yes
#59 5.285 checking for if_indextoname... yes
#59 5.380 checking for ANSI C header files... yes
#59 5.526 checking for sys/types.h... yes
#59 5.599 checking for sys/stat.h... yes
#59 5.663 checking for stdlib.h... yes
#59 5.706 checking for string.h... yes
#59 5.763 checking for memory.h... yes
#59 5.821 checking for strings.h... yes
#59 5.885 checking for inttypes.h... yes
#59 5.935 checking for stdint.h... yes
#59 5.992 checking for unistd.h... yes
#59 6.054 checking netinet/tcp.h usability... yes
#59 6.117 checking netinet/tcp.h presence... yes
#59 6.150 checking for netinet/tcp.h... yes
#59 6.167 checking sys/un.h usability... yes
#59 6.223 checking sys/un.h presence... yes
#59 6.238 checking for sys/un.h... yes
#59 6.249 checking sys/sockio.h usability... no
#59 6.300 checking sys/sockio.h presence... no
#59 6.330 checking for sys/sockio.h... no
#59 6.331 checking for field ss_family in struct sockaddr_storage... yes
#59 6.387 checking if getaddrinfo supports AI_V4MAPPED... yes
#59 6.440 checking if getaddrinfo supports AI_ALL... yes
#59 6.477 checking if getaddrinfo supports AI_IDN... no
#59 6.507 checking if ancillary credentials uses ucred... no
#59 6.543 checking if ancillary credentials uses cmsgcred... no
#59 6.630 checking for a sed that does not truncate output... /bin/sed
#59 6.638 checking for ld used by cc... /usr/bin/ld
#59 6.650 checking if the linker (/usr/bin/ld) is GNU ld... yes
#59 6.657 checking for /usr/bin/ld option to reload object files... -r
#59 6.658 checking for BSD-compatible nm... /usr/bin/nm -B
#59 6.665 checking whether ln -s works... yes
#59 6.666 checking how to recognize dependent libraries... pass_all
#59 6.708 checking dlfcn.h usability... yes
#59 6.764 checking dlfcn.h presence... yes
#59 6.778 checking for dlfcn.h... yes
#59 6.782 checking the maximum length of command line arguments... 1572864
#59 6.791 checking command to parse /usr/bin/nm -B output from cc object... ok
#59 6.945 checking for objdir... .libs
#59 6.950 checking for ar... ar
#59 6.952 checking for ranlib... ranlib
#59 6.953 checking for strip... strip
#59 7.063 checking if cc supports -fno-rtti -fno-exceptions... no
#59 7.103 checking for cc option to produce PIC... -fPIC
#59 7.106 checking if cc PIC flag -fPIC works... yes
#59 7.158 checking if cc static flag -static works... yes
#59 7.280 checking if cc supports -c -o file.o... yes
#59 7.314 checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
#59 7.341 checking whether -lc should be explicitly linked in... no
#59 7.406 checking dynamic linker characteristics... GNU/Linux ld.so
#59 7.448 checking how to hardcode library paths into programs... immediate
#59 7.450 checking whether stripping libraries is possible... yes
#59 7.457 checking if libtool supports shared libraries... yes
#59 7.458 checking whether to build shared libraries... yes
#59 7.459 checking whether to build static libraries... no
#59 7.845
#59 7.845 creating libtool
#59 7.905 appending configuration tag "CXX" to libtool
#59 8.069 configure: patching config.h.in
#59 8.076 configure: creating ./config.status
#59 8.207 config.status: creating config.h
#59 8.288 /bin/bash /usr/src/php/ext/sockets/libtool --mode=compile cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/sockets.c -o sockets.lo
#59 8.289 /bin/bash /usr/src/php/ext/sockets/libtool --mode=compile cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/multicast.c -o multicast.lo
#59 8.289 /bin/bash /usr/src/php/ext/sockets/libtool --mode=compile cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/conversions.c -o conversions.lo
#59 8.290 /bin/bash /usr/src/php/ext/sockets/libtool --mode=compile cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/sockaddr_conv.c -o sockaddr_conv.lo
#59 8.290 /bin/bash /usr/src/php/ext/sockets/libtool --mode=compile cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H  -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/sendrecvmsg.c -o sendrecvmsg.lo
#59 8.452 mkdir .libs
#59 8.454 mkdir .libs
#59 8.455 mkdir: cannot create directory '.libs': File exists
#59 8.456  cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/multicast.c  -fPIC -DPIC -o .libs/multicast.o
#59 8.457  cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/sockets.c  -fPIC -DPIC -o .libs/sockets.o
#59 8.457  cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/conversions.c  -fPIC -DPIC -o .libs/conversions.o
#59 8.462  cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/sockaddr_conv.c  -fPIC -DPIC -o .libs/sockaddr_conv.o
#59 8.463  cc -I. -I/usr/src/php/ext/sockets -I/usr/src/php/ext/sockets/include -I/usr/src/php/ext/sockets/main -I/usr/src/php/ext/sockets -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H -fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /usr/src/php/ext/sockets/sendrecvmsg.c  -fPIC -DPIC -o .libs/sendrecvmsg.o
#59 8.614 /usr/src/php/ext/sockets/sendrecvmsg.c: In function 'init_ancillary_registry':
#59 8.614 /usr/src/php/ext/sockets/sendrecvmsg.c:128:19: error: invalid application of 'sizeof' to incomplete type 'struct cmsgcred'
#59 8.614   PUT_ENTRY(sizeof(struct cmsgcred), 0, 0, from_zval_write_ucred,
#59 8.614                    ^~~~~~
#59 8.614 /usr/src/php/ext/sockets/sendrecvmsg.c:99:17: note: in definition of macro 'PUT_ENTRY'
#59 8.614   entry.size   = sizev; \
#59 8.614                  ^~~~~
#59 8.617 /usr/src/php/ext/sockets/sendrecvmsg.c:129:36: error: 'SCM_CREDS' undeclared (first use in this function); did you mean 'SCM_SRCRT'?
#59 8.617     to_zval_read_ucred, SOL_SOCKET, SCM_CREDS);
#59 8.617                                     ^~~~~~~~~
#59 8.617 /usr/src/php/ext/sockets/sendrecvmsg.c:105:19: note: in definition of macro 'PUT_ENTRY'
#59 8.617   key.cmsg_type  = type; \
#59 8.617                    ^~~~
#59 8.617 /usr/src/php/ext/sockets/sendrecvmsg.c:129:36: note: each undeclared identifier is reported only once for each function it appears in
#59 8.617     to_zval_read_ucred, SOL_SOCKET, SCM_CREDS);
#59 8.617                                     ^~~~~~~~~
#59 8.617 /usr/src/php/ext/sockets/sendrecvmsg.c:105:19: note: in definition of macro 'PUT_ENTRY'
#59 8.617   key.cmsg_type  = type; \
#59 8.617                    ^~~~
#59 8.630 make: *** [Makefile:216: sendrecvmsg.lo] Error 1
#59 8.630 make: *** Waiting for unfinished jobs....
#59 ERROR: executor failed running [/bin/sh -c install-php-extensions sockets]: exit code: 2
------
 > [build-version-8015 47/68] RUN install-php-extensions sockets:
------
executor failed running [/bin/sh -c install-php-extensions sockets]: exit code: 2
```

**Comment by llaville** [comment]
@mlocati it's confirmed. Found PHP report issue at https://github.com/php/php-src/issues/7978

**Comment by llaville** [comment]
@mlocati Issue was fixed in php-src repository. As It affect Docker images of PHP 8.0.15 and 8.1.2, either we wait new docker image releases to confirm or close this report now; I'll let you decide what it's better !

**Comment by jkobus** [comment]
For anyone out there who is having the same issue: just use the 8.0.14 version.

**Comment by mlocati** [comment]
I've just published version 1.4.13 which should fix this issue

**Comment by llaville** [comment]
Thanks, will have a look tomorrow !

**Comment by llaville** [comment]
@mlocati Thanks a lot ❤️ for your fix (v1.4.13). I confirm that its works fine for PHP 8.0.15 and 8.1.2

**Comment by mlocati** [comment]
Thanks for checking, Laurent!
